openmdao/solvers/linear/direct.py

```diff
@@ -82,16 +91,20 @@ def _mat_vec(self, in_vec, out_vec):
         system = self._system
         mode = self._mode
 
-        # assign x and b vectors based on fwd mode
-        x_vec = system._vectors['output'][vec_name]
-        b_vec = system._vectors['residual'][vec_name]
+        # assign x and b vectors based on mode
```

Comment:
I don't understand why this is necessary. You should be able to assemble the jacobian via fwd products regardless. 

Why do we need mode?

Reply:
After discussion with Justin, we decided to always assemble the Jac in forward mode, like Newton, so the change to the _linearize API is unneeded.